private/check/check.go

```diff
@@ -40,9 +46,33 @@ func (g *TestFixtureGetter) LoadTestFixture(path string) (*verify.TestFixture, e
 
 		g.cache[path] = fixture
 	}
+
 	return fixture, nil
 }
 
+type TestFixtureCtx struct {
+	Fixture *verify.TestFixture
+	Path    string
+}
+
+func (g *TestFixtureGetter) LoadAll() <-chan *TestFixtureCtx {
```

Comment:
`LoadAll` sounds very similar to `Load` but it's doing something quite different. Shall we call it `GetAll` or something along those lines?

Also, since the hard work has already been done to populate the cache, is there any advantage to using a channel and a goroutine here? Can't we simply iterate over the cache, build an array of `TestFixtureCtx` and return it?

Reply:
Happy to return a slice over an array. I figured this might be slightly more favourable in that the memory requirements might be less than a full array. But maybe there are other overheads I'm not familiar with.

Reply:
You have a point there. If the other end simply loops over the channel and builds up a collection of fixtures, then that memory is going to be used anyway (along with at least 2kb for the goroutine stack + synchronization overheads). If it's going to work on each item as it arrives over the channel and moves on to the next, then that might be more memory efficient _if_ there are many fixtures. For a small number of fixtures, my hunch is that goroutine scheduling overhead dominates everything. Hard to say definitely without some benchmarks though.

Reply:
This prompted me to do some reading too (for my own curiosity) and I realised the same conclusion! Array makes loads more sense, about to push this and other changes for the comments.